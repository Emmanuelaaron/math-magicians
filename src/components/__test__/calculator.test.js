import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../calculator';

describe('Calculator', () => {
  it('should render the calculator component', () => {
    const calc = renderer.create(<Calculator />);

    expect(calc).toMatchSnapshot()
  });
});
