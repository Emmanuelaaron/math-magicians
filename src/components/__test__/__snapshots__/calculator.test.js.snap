// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render the calculator component 1`] = `
<div
  className="calculator"
>
  <p>
    Lets do some Mathematics
  </p>
  <div
    className="row border w-50"
  >
    <div
      className="col-12 div height border d-flex justify-content-end align-items-end bg-secondary text-light"
    >
      0
    </div>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center bg-warning"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center bg-warning"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center bg-warning"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center bg-warning"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="col-6 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="col-3 div height border d-flex justify-content-center align-items-center bg-warning"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
